declarations_site/catalog/static/js/main.js

```diff
@@ -58,6 +58,27 @@ $(function() {
         $(this).toggleClass('open');
     });
 
+    $('#login-modal').on('shown.bs.modal', function (e) {
+        var href = $(e.relatedTarget).data('href');
+        if (href) {
+            href = href.replace('?', '%3F').replace('&', '%26');
+            $(e.currentTarget).find('#signin a').each(function (){
+                this.href = this.href.replace('?next=', '?next='+href+'&origin=');
+            });
+        }
+    });
+
+    function _showLoginModal() {
+        var href = window.location.href,
+            pos = href.search('login_to=');
+        if (pos > 0)
+            href = href.substr(pos+9);
```

Comment:
Hardcoded value. Better to store the string in a variable and then use it and it's lenght